From cd760628dbcd46e3a6ddc837aaa8d06313001490 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Fri, 28 Feb 2025 12:35:57 +0100
Subject: [PATCH] Regex patch (#35)

* Regex patch

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
---
 pyiron_vasp/vasp/structure.py | 3 ++-
 pyiron_vasp/vasp/vasprun.py   | 4 +++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/pyiron_vasp/vasp/structure.py b/pyiron_vasp/vasp/structure.py
index 1d0a8d9..c08c7af 100644
--- a/pyiron_vasp/vasp/structure.py
+++ b/pyiron_vasp/vasp/structure.py
@@ -3,6 +3,7 @@
 # Distributed under the terms of "New BSD License", see the LICENSE file.
 
 import os
+import re
 from collections import OrderedDict
 from ase.atoms import Atoms
 from ase.constraints import FixCartesian
@@ -393,7 +394,7 @@ def _dict_to_atoms(atoms_dict, species_list=None, read_from_first_line=False):
     elements_new = list()
     for ele in elements:
         for e in ele:
-            elements_new.append(e)
+            elements_new.append(re.split("[^a-zA-Z]", e)[0])
     elements = elements_new
     if is_absolute:
         atoms = Atoms(elements, positions=positions, cell=cell, pbc=True)
diff --git a/pyiron_vasp/vasp/vasprun.py b/pyiron_vasp/vasp/vasprun.py
index f0b8cab..0e9334e 100644
--- a/pyiron_vasp/vasp/vasprun.py
+++ b/pyiron_vasp/vasp/vasprun.py
@@ -3,6 +3,7 @@
 # Distributed under the terms of "New BSD License", see the LICENSE file.
 
 import os
+import re
 
 from ase.atoms import Atoms
 from ase.constraints import FixCartesian
@@ -646,7 +647,8 @@ def get_initial_structure(self):
         """
         # try:
         el_list = [
-            el.split("_")[0] for el in self.vasprun_dict["atominfo"]["species_list"]
+            re.split("[^a-zA-Z]", el)[0]
+            for el in self.vasprun_dict["atominfo"]["species_list"]
         ]
         cell = self.vasprun_dict["init_structure"]["cell"]
         positions = self.vasprun_dict["init_structure"]["positions"]
